### PR TITLE
Bug/add support for wsl2

### PR DESCRIPTION
**Pull request by aseaday**
https://github.com/tensorchord/envd/issues/1133

### CODE REVIEW

File: pkg/envd/docker.go [hunk 1]
```diff
@@ -581,7 +581,14 @@ func (e dockerEngine) GPUEnabled(ctx context.Context) (bool, error) {
 	}
 	logrus.WithField("info", info).Debug("docker info")
 	nv := info.Runtimes["nvidia"]
-	return nv.Path != "", nil
+	if nv.Path != "" {
+		return true, nil
+	} else if strings.HasSuffix(info.KernelVersion, "WSL2") {
```
Comment (gaocegege) [inline pkg/envd/docker.go:586]: /cc @VoVAllen

File: pkg/envd/docker.go [hunk 2]
```diff
@@ -581,7 +581,14 @@ func (e dockerEngine) GPUEnabled(ctx context.Context) (bool, error) {
...
+		logrus.Warn("We couldn't detect if your runtime support GPU on WSL2, we will continue to run your environment.")
+		return true, nil
```
Comment (VoVAllen) [inline pkg/envd/docker.go:588]: What if user doesn't have GPU with WSL?

Comment (aseaday) [reply pkg/envd/docker.go:588]: I raised a issue for docker win team how to detect cuda device.
But for now, users can start environment but without cuda support

**Review by VoVAllen — APPROVED** [review]
/lgtm

**Pull request merged**